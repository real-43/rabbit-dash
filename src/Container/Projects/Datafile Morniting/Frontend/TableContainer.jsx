import React, { Fragment, useState } from 'react';
import {
  useTable,
  useSortBy,
  useFilters,
  useExpanded,
  usePagination, 
} from 'react-table';
import { Table, Row, Col, Button, Input } from 'reactstrap';
import { Filter, DefaultColumnFilter } from './filters';
import './Table.css'
import jsPDF from "jspdf";
import "jspdf-autotable";
import Loading from '../../../../components/Loading';

const TableContainer = ({ columns, data, renderRowSubComponent, genPDF }) => {

  const [isLoading, setIsLoading] = useState(false);

  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    page,
    prepareRow,
    visibleColumns,
    canPreviousPage,
    canNextPage,
    pageOptions,
    pageCount,
    gotoPage,
    nextPage,
    previousPage,
    setPageSize,
    state: { pageIndex, pageSize },
  } = useTable(
    {
      columns,
      data,
      defaultColumn: { Filter: DefaultColumnFilter },
      initialState: { pageIndex: 0, pageSize: 10 },
    },
    useFilters,
    useSortBy,
    useExpanded,
    usePagination
  );

  const generateSortingIndicator = (column) => {
    return column.isSorted ? (column.isSortedDesc ? ' ðŸ”½' : ' ðŸ”¼') : '';
  };

  const onChangeInSelect = (event) => {
    setPageSize(Number(event.target.value));
  };

  function exportPDF() {
    setIsLoading(true)
    console.log("loading", isLoading)
    const unit = "pt";
    const size = "A3"; // Use A1, A2, A3 or A4
    const orientation = "portrait"; // portrait or landscape

    const marginLeft = 40;
    const doc = new jsPDF(orientation, unit, size);

    doc.setFontSize(15);

    const title = "Datafile Mornitoring";
    const headers = [["ID", "DEVICE ID", "LOCATION ID", "SOURCE TYPE", "STATUS", "TXN DATE", "TXN TYPE", "VALUE", "CREATE DATA AT", "UPDATE DATA AT"]];

    const info = []
    page.map((p, index) => {
      let arr = []
      p.cells.map((cell) => {
        arr.push(cell.value)
      })
      info.push(arr)
    })

    let content = {
        startY: 50,
        head: headers,
        body: info
    };

    doc.text(title, marginLeft, 40);
    doc.autoTable(content);
    doc.save("report.pdf")
    setIsLoading(false)
    console.log("loading", isLoading)
  }

  return (
    <div className='content-wrapper'>
      <Loading isLoading={isLoading} />
      <div className='pdf-page'>
        <div className='pdf-table'>
          <Fragment >
              <Table bordered hover {...getTableProps()} >
                  <thead className='header-table'>
                      {headerGroups.map((headerGroup) => (
                          <tr {...headerGroup.getHeaderGroupProps()}>
                              {headerGroup.headers.map((column) => (
                                  <th {...column.getHeaderProps()} style={{width: "fit-content"}}>
                                      <div className='header-content' {...column.getSortByToggleProps()}>
                                          {column.render('Header')}
                                          {generateSortingIndicator(column)}
                                      </div>
                                      <Filter column={column} />
                                  </th>
                              ))}
                          </tr>
                      ))}
                  </thead>
                  <tbody {...getTableBodyProps()}>
                      {page.map((row) => {
                          prepareRow(row);
                          return (
                                  <Fragment key={row.getRowProps().key}>
                                      <tr>
                                          {row.cells.map((cell) => {
                                              return (
                                                  <td className='content-tabel' {...cell.getCellProps()}>{cell.render('Cell')}</td>
                                              );
                                          })}
                                      </tr>
                                  </Fragment>
                              );
                          })}
                  </tbody>
              </Table>

            <Row className='under-table'>
              <Col md={3}>
                <Button
                  color='primary'
                  onClick={() => gotoPage(0)}
                  disabled={!canPreviousPage}
                >
                  {'<<'}
                </Button>
                <Button
                  color='primary'
                  onClick={previousPage}
                  disabled={!canPreviousPage}
                >
                  {'<'}
                </Button>
              </Col>
              <Col md={2} style={{ marginTop: 7 }}>
                Page{' '}
                <strong>
                  {pageIndex + 1} of {pageOptions.length}
                </strong>
              </Col>
              <Col md={2}>
                <Input
                  type='select'
                  value={pageSize}
                  onChange={onChangeInSelect}
                >
                  {'>'}
                  {[10, 20, 30, 40, 50, 10**10].map((pageSize) => (
                    <option key={pageSize} value={pageSize}>
                      Show {(pageSize === 10**10) ? "All" : `${pageSize}`}
                    </option>
                  ))}
                </Input>
              </Col>
              <Col md={3}>
                <Button color='primary' onClick={nextPage} disabled={!canNextPage}>
                  {'>'}
                </Button>
                <Button
                  color='primary'
                  onClick={() => gotoPage(pageCount - 1)}
                  disabled={!canNextPage}
                >
                  {'>>'}
                </Button>
              </Col>
            </Row>
          </Fragment>
        </div>
        {(genPDF.genPDF) ? (
          <div>
            <button className="pdf-btn" onClick={(e) => exportPDF()}>Generate showing data to PDF</button>
          </div>
        ) : ""}
      </div>
    </div>
  );
};

export default TableContainer;