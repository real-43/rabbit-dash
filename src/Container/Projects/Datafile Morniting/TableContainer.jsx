import React, { Fragment } from 'react';
import {
  useTable,
  useSortBy,
  useFilters,
  useExpanded,
  usePagination, 
} from 'react-table';
import { Table, Row, Col, Button, Input } from 'reactstrap';
import { Filter, DefaultColumnFilter } from './filters';
import './Table.css'

const TableContainer = ({ columns, data, renderRowSubComponent }) => {
  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    page,
    prepareRow,
    visibleColumns,
    canPreviousPage,
    canNextPage,
    pageOptions,
    pageCount,
    gotoPage,
    nextPage,
    previousPage,
    setPageSize,
    state: { pageIndex, pageSize },
  } = useTable(
    {
      columns,
      data,
      defaultColumn: { Filter: DefaultColumnFilter },
      initialState: { pageIndex: 0, pageSize: 10 },
    },
    useFilters,
    useSortBy,
    useExpanded,
    usePagination
  );

  const generateSortingIndicator = (column) => {
    return column.isSorted ? (column.isSortedDesc ? ' ðŸ”½' : ' ðŸ”¼') : '';
  };

  const onChangeInSelect = (event) => {
    setPageSize(Number(event.target.value));
  };

  return (
    <Fragment >
        <Table bordered hover {...getTableProps()} >
            <thead>
                {headerGroups.map((headerGroup) => (
                    <tr {...headerGroup.getHeaderGroupProps()}>
                        {headerGroup.headers.map((column) => (
                            <th {...column.getHeaderProps()} style={{width: "fit-content"}}>
                                <div {...column.getSortByToggleProps()}>
                                    {column.render('Header')}
                                    {generateSortingIndicator(column)}
                                </div>
                                <Filter column={column} />
                            </th>
                        ))}
                    </tr>
                ))}
            </thead>

            <tbody {...getTableBodyProps()}>
                {page.map((row) => {
                    prepareRow(row);
                    return (
                            <Fragment key={row.getRowProps().key}>
                                <tr>
                                    {row.cells.map((cell) => {
                                        return (
                                            <td {...cell.getCellProps()}>{cell.render('Cell')}</td>
                                        );
                                    })}
                                </tr>
                                {row.isExpanded && (
                                    <tr>
                                        <td colSpan={visibleColumns.length}>
                                            {renderRowSubComponent(row)}
                                        </td>
                                    </tr>
                                )}
                            </Fragment>
                        );
                    })}
            </tbody>
        </Table>

      <Row className='under-table'>
        <Col md={3}>
          <Button
            color='primary'
            onClick={() => gotoPage(0)}
            disabled={!canPreviousPage}
          >
            {'<<'}
          </Button>
          <Button
            color='primary'
            onClick={previousPage}
            disabled={!canPreviousPage}
          >
            {'<'}
          </Button>
        </Col>
        <Col md={2} style={{ marginTop: 7 }}>
          Page{' '}
          <strong>
            {pageIndex + 1} of {pageOptions.length}
          </strong>
        </Col>
        <Col md={2}>
          <Input
            type='select'
            value={pageSize}
            onChange={onChangeInSelect}
          >
            {'>'}
            {[10, 20, 30, 40, 50].map((pageSize) => (
              <option key={pageSize} value={pageSize}>
                Show {pageSize}
              </option>
            ))}
          </Input>
        </Col>
        <Col md={3}>
          <Button color='primary' onClick={nextPage} disabled={!canNextPage}>
            {'>'}
          </Button>
          <Button
            color='primary'
            onClick={() => gotoPage(pageCount - 1)}
            disabled={!canNextPage}
          >
            {'>>'}
          </Button>
        </Col>
      </Row>
    </Fragment>
  );
};

export default TableContainer;