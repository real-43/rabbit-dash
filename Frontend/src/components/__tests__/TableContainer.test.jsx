import React from 'react';
import TableContainer from '../../Container/Projects/Datafile Monitoring/Frontend/TableContainer';
import { data } from '../../Mockup Data/Data';
import { create } from 'react-test-renderer'

test("TableContainerComponent render", () => {

    const columns =  [
        {
            Header: 'ID',
            accessor: 'id',
            Filter: 'NoSearch',
        },
        {
            Header: 'Device ID',
            accessor: 'deviceId',
        },
        {
            Header: 'Location ID',
            accessor: 'locationId',
        },
        {
            Header: 'Source Type',
            accessor: 'sourceType',
            disableSortBy: true,
            Filter: "SelectColumnFilter",
            filter: 'equals',
        },
        {
            Header: "Status",
            accessor: 'status',
            disableSortBy: true,
            Filter: 'SelectColumnFilter',
            filter: 'equals',
        },
        {
            Header: () => {
                return (
                <div className="header-txn-date">
                    <p>Txn Date</p>
                    <button onClick={() => window.location.reload(false)}><i className='fas fa-times'></i></button>
                </div>
                )
            },
            accessor: 'txnDate',
            disableSortBy: true,
            Filter: 'DateColumnFilter',
        },
        {
            Header: "Txn Type",
            accessor: 'txnType',
            disableSortBy: true,
            Filter: 'SelectColumnFilter',
            filter: 'equals',
        },
        {
            Header: 'Value',
            accessor: 'value',
        },
        {
            Header: 'Create Date',
            accessor: 'createdAt',
            Filter: 'CNUDateColumnFilter',
        },
        {
            Header: 'Update Date',
            accessor: 'updatedAt',
            Filter: 'CNUDateColumnFilter',
        },
    ]
    const genPDF = false
    const wrapper = create(<TableContainer columns={columns} data={data} genPDF={genPDF}/>).toJSON();
    expect(wrapper).toMatchSnapshot()
})