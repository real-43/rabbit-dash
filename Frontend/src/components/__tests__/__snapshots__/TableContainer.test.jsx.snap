// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContainerComponent render 1`] = `
<div
  className="content-wrapper"
>
  
  <div
    className="pdf-page"
  >
    <div
      className="pdf-table"
    >
      <table
        className="table table-bordered table-hover"
        role="table"
      >
        <thead
          className="header-table"
        >
          <tr
            role="row"
          >
            <th
              colSpan={1}
              role="columnheader"
              style={
                Object {
                  "width": "fit-content",
                }
              }
            >
              <div
                className="header-content"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                ID
                
              </div>
              <div
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                NoSearch
              </div>
            </th>
            <th
              colSpan={1}
              role="columnheader"
              style={
                Object {
                  "width": "fit-content",
                }
              }
            >
              <div
                className="header-content"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Device ID
                
              </div>
              <div
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="search..."
                  style={
                    Object {
                      "height": "30px",
                      "marginLeft": "auto",
                      "marginRight": "auto",
                      "width": "70px",
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </th>
            <th
              colSpan={1}
              role="columnheader"
              style={
                Object {
                  "width": "fit-content",
                }
              }
            >
              <div
                className="header-content"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Location ID
                
              </div>
              <div
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="search..."
                  style={
                    Object {
                      "height": "30px",
                      "marginLeft": "auto",
                      "marginRight": "auto",
                      "width": "70px",
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </th>
            <th
              colSpan={1}
              role="columnheader"
              style={
                Object {
                  "width": "fit-content",
                }
              }
            >
              <div
                className="header-content"
                style={
                  Object {
                    "cursor": undefined,
                  }
                }
              >
                Source Type
                
              </div>
              <div
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                SelectColumnFilter
              </div>
            </th>
            <th
              colSpan={1}
              role="columnheader"
              style={
                Object {
                  "width": "fit-content",
                }
              }
            >
              <div
                className="header-content"
                style={
                  Object {
                    "cursor": undefined,
                  }
                }
              >
                Status
                
              </div>
              <div
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                SelectColumnFilter
              </div>
            </th>
            <th
              colSpan={1}
              role="columnheader"
              style={
                Object {
                  "width": "fit-content",
                }
              }
            >
              <div
                className="header-content"
                style={
                  Object {
                    "cursor": undefined,
                  }
                }
              >
                <div
                  className="header-txn-date"
                >
                  <p>
                    Txn Date
                  </p>
                  <button
                    onClick={[Function]}
                  >
                    <i
                      className="fas fa-times"
                    />
                  </button>
                </div>
                
              </div>
              <div
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                DateColumnFilter
              </div>
            </th>
            <th
              colSpan={1}
              role="columnheader"
              style={
                Object {
                  "width": "fit-content",
                }
              }
            >
              <div
                className="header-content"
                style={
                  Object {
                    "cursor": undefined,
                  }
                }
              >
                Txn Type
                
              </div>
              <div
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                SelectColumnFilter
              </div>
            </th>
            <th
              colSpan={1}
              role="columnheader"
              style={
                Object {
                  "width": "fit-content",
                }
              }
            >
              <div
                className="header-content"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Value
                
              </div>
              <div
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="search..."
                  style={
                    Object {
                      "height": "30px",
                      "marginLeft": "auto",
                      "marginRight": "auto",
                      "width": "70px",
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </th>
            <th
              colSpan={1}
              role="columnheader"
              style={
                Object {
                  "width": "fit-content",
                }
              }
            >
              <div
                className="header-content"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Create Date
                
              </div>
              <div
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                CNUDateColumnFilter
              </div>
            </th>
            <th
              colSpan={1}
              role="columnheader"
              style={
                Object {
                  "width": "fit-content",
                }
              }
            >
              <div
                className="header-content"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Update Date
                
              </div>
              <div
                style={
                  Object {
                    "marginTop": 5,
                  }
                }
              >
                CNUDateColumnFilter
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr>
            <td
              className="content-tabel"
              role="cell"
            >
              1
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              000A22E8
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              244
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              LP
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              unmatched
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              20/02/2021 07:49:21
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              Purse Use
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              116.75
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.467Z
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.499Z
            </td>
          </tr>
          <tr>
            <td
              className="content-tabel"
              role="cell"
            >
              2
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              000A22E8
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              244
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              LP
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              unmatched
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              15/02/2021 13:49:59
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              Purse Use
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              317.00
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.467Z
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.499Z
            </td>
          </tr>
          <tr>
            <td
              className="content-tabel"
              role="cell"
            >
              3
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              000A22E8
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              244
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              LP
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              unmatched
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              11/02/2021 07:18:52
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              Purse Use
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              42.00
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.467Z
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.499Z
            </td>
          </tr>
          <tr>
            <td
              className="content-tabel"
              role="cell"
            >
              4
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              000A4D16
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              16777215
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              LP
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              unmatched
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              17/01/2021 17:46:15
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              Purse Use
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              150.00
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.467Z
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.499Z
            </td>
          </tr>
          <tr>
            <td
              className="content-tabel"
              role="cell"
            >
              5
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              000A4253
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              16777215
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              LP
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              unmatched
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              24/02/2021 18:55:20
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              Purse Use
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              178.25
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.467Z
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.499Z
            </td>
          </tr>
          <tr>
            <td
              className="content-tabel"
              role="cell"
            >
              6
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              000A4253
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              16777215
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              LP
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              unmatched
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              25/02/2021 20:23:33
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              Purse Use
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              135.00
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.467Z
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.499Z
            </td>
          </tr>
          <tr>
            <td
              className="content-tabel"
              role="cell"
            >
              7
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              000A4253
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              16777215
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              LP
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              unmatched
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              25/02/2021 21:49:41
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              Purse Use
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              25.00
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.467Z
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.499Z
            </td>
          </tr>
          <tr>
            <td
              className="content-tabel"
              role="cell"
            >
              8
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              000A4253
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              16777215
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              LP
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              unmatched
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              25/02/2021 22:03:00
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              Purse Use
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              46.00
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.467Z
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.499Z
            </td>
          </tr>
          <tr>
            <td
              className="content-tabel"
              role="cell"
            >
              9
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              000A22E8
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              244
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              LP
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              unmatched
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              24/02/2021 19:25:34
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              Purse Use
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              243.00
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.467Z
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.499Z
            </td>
          </tr>
          <tr>
            <td
              className="content-tabel"
              role="cell"
            >
              10
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              000A22E8
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              244
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              LP
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              unmatched
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              24/02/2021 20:22:24
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              Purse Use
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              56.00
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.467Z
            </td>
            <td
              className="content-tabel"
              role="cell"
            >
              2021-05-27T11:20:27.499Z
            </td>
          </tr>
        </tbody>
      </table>
      <div
        className="under-table row"
      >
        <div
          className="col-md-3"
        >
          <button
            className="btn btn-primary disabled"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            &lt;&lt;
          </button>
          <button
            className="btn btn-primary disabled"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            &lt;
          </button>
        </div>
        <div
          className="col-md-2"
          style={
            Object {
              "marginTop": 7,
            }
          }
        >
          Page
           
          <strong>
            1
             of 
            5
          </strong>
        </div>
        <div
          className="col-md-2"
        >
          <select
            className="form-select"
            onChange={[Function]}
            value={10}
          >
            &gt;
            <option
              value={10}
            >
              Show 
              10
            </option>
            <option
              value={20}
            >
              Show 
              20
            </option>
            <option
              value={30}
            >
              Show 
              30
            </option>
            <option
              value={40}
            >
              Show 
              40
            </option>
            <option
              value={50}
            >
              Show 
              50
            </option>
            <option
              value={10000000000}
            >
              Show 
              All
            </option>
          </select>
        </div>
        <div
          className="col-md-3"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            &gt;
          </button>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            &gt;&gt;
          </button>
        </div>
      </div>
    </div>
    
  </div>
</div>
`;
